FinSearchUnified/Subscriber/Frontend.php

```diff
@@ -190,33 +219,55 @@ public function onAjaxSearchIndexAction()
     }
 
     /**
-     * @param Request $request
+     * @param Enlight_Controller_Request_Request $request
      *
      * @return bool
      */
-    private function isSearchPage(Request $request)
+    private function isSearchPage(Enlight_Controller_Request_Request $request)
     {
         return array_key_exists('sSearch', $request->getParams());
     }
 
     /**
-     * @param Request $request
+     * @param Enlight_Controller_Request_Request $request
      *
      * @return bool
      */
-    private function isCategoryPage(Request $request)
+    private function isCategoryPage(Enlight_Controller_Request_Request $request)
     {
         return $request->getControllerName() === 'listing' && $request->getActionName() !== 'manufacturer' &&
             array_key_exists('sCategory', $request->getParams());
     }
 
     /**
-     * @param Request $request
+     * @param Enlight_Controller_Request_Request $request
      *
      * @return bool
      */
-    private function isManufacturerPage(Request $request)
+    private function isManufacturerPage(Enlight_Controller_Request_Request $request)
     {
         return $request->getControllerName() === 'listing' && $request->getActionName() === 'manufacturer';
     }
+
+    /**
+     * @param Enlight_Controller_Request_Request $request
+     *
+     * @return bool
+     */
+    public function isWhiteListed(Enlight_Controller_Request_Request $request)
+    {
+        $controllerName = strtolower($request->getControllerName());
+
+        foreach (self::WHITE_LIST as $name => $actions) {
+            if (!is_array($actions)) {
```

Comment:
I think these condition checks can be optimized:
```PHP
return $controllerName === $actions || ($controllerName === $name && in_array($request->getActionName(), $actions));
```
Also add a comment explaining the conditions.

Reply:
Simplifying this wont work as we need to check if any value returns true. Simplifying this would also return the first false encountered which is not intended behavior.

Reply:
You are right. I forgot about the loop. How about
```PHP
if ($controllerName === $actions || ($controllerName === $name && in_array($request->getActionName(), $actions))) {
    return true;
}
```
?

Reply:
ah, yes cant argue with this  :wink: